qpid-jms-client/src/test/java/org/apache/qpid/jms/transports/netty/NettyTcpTransportTest.java

```diff
@@ -730,14 +821,30 @@ private void assertEpoll(String message, boolean expected, Transport transport)
             }
         }
 
-        assertNotNull("Transport implementation unknown", group);
+        assertNotNull("Transport implementation unknown", groupRefField);
 
-        group.setAccessible(true);
-        if (expected) {
-            assertTrue(message, group.get(transport) instanceof EpollEventLoopGroup);
-        } else {
-            assertFalse(message, group.get(transport) instanceof EpollEventLoopGroup);
+        groupRefField.setAccessible(true);
+        return (EventLoopGroupRef) groupRefField.get(transport);
+    }
+
+    private static void assertEventLoopGroupType(String message, Transport transport) throws Exception {
+        final EventLoopGroupRef groupRef = getGroupRef(transport);
+        Class<? extends EventLoopGroup> eventLoopGroupClass = null;
+        switch (groupRef.type()) {
+
+            case EPOLL:
+                eventLoopGroupClass = EpollEventLoopGroup.class;
+                break;
+            case KQUEUE:
+                eventLoopGroupClass = KQueueEventLoopGroup.class;
+                break;
+            case NIO:
+                eventLoopGroupClass = NioEventLoopGroup.class;
+                break;
+            default:
+                fail("Unsupported type: " + groupRef.type());
```

Comment:
This seemingly entirely breaks the usefullness of the assertion. Previously it asserted the group type matched expectation. Now it essentially asserts the the 'getType' value of the 'Ref' aligns with the type of the 'ref()' group value, but does no verification that either of those are correct according to expectations, i.e it would still pass with entirely the wrong group type being returned, defeating the whole point of the original test.

If you want to combine the old methods into something general, pass in the *expected* value(s) so it can do what it was doing, as well as also check the Ref type and group values align.

Reply:
let me change this, right!